lib/status/status.js

```diff
@@ -71,6 +71,42 @@ class Status {
     }
   }
 
+  async cleanupLocalBuilds(repoName, refMode, ref) {
+    let { logger } = this
+
+    logger.info(
+      `cleaning up local builds for ${refMode} ${ref} on ${repoName}`,
+      { module: 'status' }
+    )
+
+    let now = Date.now()
+
+    await this._updateRepoStatus(repoName, async(repoStatus) => {
+      for (let buildId in repoStatus.builds) {
+        let build = repoStatus.builds[buildId]
+        let { branch, tag, status, extra } = build
+
+        if (
+          ((refMode === 'branch' && branch === ref)
+            || (refMode === 'tag' && tag === ref))
+          && status == 'success'
+          && extra
+          && extra.localDirectory
+          && !extra.localCleaned
+        ) {
+          logger.debug(
+            `removing local build ${buildId} at ${extra.localDirectory}`,
+            { module: 'status' }
+          )
+
+          await remove(extra.localDirectory)
+          build.updated = now
+          extra.localCleaned = true
+        }
+      }
+    })
+  }
+
```

Comment:
this is pretty straightforward: just lookup a build with that branch/tag and extract its local deployment directory, then remove it and add a cleaned marker